torch/csrc/utils/python_numbers.h

```diff
@@ -100,25 +105,17 @@ inline bool THPUtils_unpackBool(PyObject* obj) {
 }
 
 inline bool THPUtils_checkDouble(PyObject* obj) {
-  bool is_numpy_scalar;
+  bool is_numpy_double = false;
 #ifdef USE_NUMPY
-  is_numpy_scalar = torch::utils::is_numpy_scalar(obj);
-#else
-  is_numpy_scalar = false;
-#endif
-#if PY_MAJOR_VERSION == 2
-  return PyFloat_Check(obj) || PyLong_Check(obj) || PyInt_Check(obj) || is_numpy_scalar;
-#else
-  return PyFloat_Check(obj) || PyLong_Check(obj) || is_numpy_scalar;
+  is_numpy_double = torch::utils::is_numpy_scalar(obj) && ! torch::utils::is_numpy_int(obj);
```

Comment:
What does this do for `np.bool`? Is this true?

Reply:
np.bool returns false, consistent with the original implementation of checkLong above which for python objects is `PyLong_Check(obj) && !PyBool_Check(obj)`

Reply:
So np.int64 is not a numpy double, but np.bool is a numpy double for the purposes of this function? The naming seems off and I wonder what consequences this has

Reply:
reverted the behavioral changes back to the meaning of "castable," so now it's `scalar && ! bool`.